### PR TITLE
Add next-head-count & remove next-head class

### LINKED ISSUE #3494: 'next-head'  class being added to all the items in the the <HEAD>
I am using HEAD to populate the tags in the HTML document. Every tag though is being populated with a class added to it 'next-head'. Google SEO isn't liking it. How can i remove it?

screen shot 2017-12-23 at 9 31 59 pm

### PR DESCRIPTION
**Pull request by devknoll**
Fixes #3494

Removes `class="next-head"` from the children of the `<Head>` component. Instead, a single sentinel meta element named `next-head-count` is appended. The content is the number of contiguous elements immediately preceding the sentinel that _would have had_ the `class="next-head"` attribute.

During an update, instead of searching for `class="next-head"`, the sentinel is located and the N previous elements are considered candidates for `oldTags`. New elements are inserted before the sentinel, and finally the sentinel is updated to reflect the new count.

### CODE REVIEW

**Review by timneutkens — COMMENTED** [review]
I wonder if the bundle size increase is needed 🤔

**Comment by devknoll** [comment]
Ping @timneutkens. Not sure why these changes would affect the size like that. Any ideas?

**Comment by Timer** [comment]
@devknoll looks like a polyfill for `parseInt` is being added.

**Comment by devknoll** [comment]
D'oh... I could also just use the `Number` constructor, since I'm not relying on any of the other behavior of `parseInt`.

File: packages/next/client/head-manager.js [hunk 1]
```diff
@@ -46,9 +46,17 @@ export default class HeadManager {
...
     const headEl = document.getElementsByTagName('head')[0]
-    const oldTags = Array.prototype.slice.call(
-      headEl.querySelectorAll(type + '.next-head')
-    )
+    const headCountEl = headEl.querySelector('meta[name=next-head-count]')
+    const headCount = Number(headCountEl.content)
+    const oldTags = []
+
+    let currentEl = headCountEl.previousElementSibling
+    for (let i = 0; i < headCount; i++) {
```
Comment (Timer) [inline packages/next/client/head-manager.js]: Looks like we can save a few bytes if this is renamed to `j`.

Comment (devknoll) [reply packages/next/client/head-manager.js]: Sorry, GitHub doesn't make it easy to see exactly what you're referring to. Do you mean `i`, `headCount`, or `currentEl` should be `j`? In the case of `headCount`, it's being used later.

Comment (Timer) [reply packages/next/client/head-manager.js]: I specifically meant (prev `j`, but it's now used):
```suggestion
    for (let k = 0; k < headCount; k++) {
```

**Comment by devknoll** [comment]
The diff view from the bot is super handy BTW, thanks!

**Review by Timer — APPROVED** [review]

**Comment by Timer** [comment]
Thanks for these changes, @devknoll!

**Comment by chuyik** [comment]
This change brings issue https://github.com/zeit/next.js/issues/8220

**Pull request merged**